src/dwarf2pdb.cpp

```diff
@@ -716,7 +751,7 @@ bool CV2PDB::addDWARFProc(DWARF_InfoData& procid, DWARF_CompilationUnit* cu, DIE
 		int off = 8;
 
 		DIECursor prev = cursor;
-		while (cursor.readNext(id, true) && id.tag == DW_TAG_formal_parameter)
+		while (cursor.readNext(id, true))

```

Comment:
This changes the loop and the result of `prev`. Was this a bug in the implementation?

Reply:
This change (and the one right after) come from the fact that DWARF allows a `DW_TAG_subprogram` DIE to have the following sequence of children:

```
DW_TAG_subprogram
    DW_TAG_array_type
    DW_TAG_formal_parameter
    DW_TAG_variable
    DW_TAG_formal_parameter
```
My understanding is that the previous code assumed that the first children had to be `DW_TAG_formal_parameter` and only then the other children (none of which could be `DW_TAG_formal_parameter again`). So this change and the one below are a kind of generalization: we go through all children once to look for parameters, and then a second time for lexical blocks and their variables, so that we can handle “inerleaves” children.